### PR TITLE
Setup multi wallet support for Flutter Client library and example app

### PR DESCRIPTION
**Pull request by AMITGOELNY**
### _Summary_

Setup multi wallet support for Flutter Client library and example app

### _How did you test your changes?_

https://user-images.githubusercontent.com/4512097/204870763-b57e8df2-68fc-4b45-8ce5-3bc00a8cb753.mov

### CODE REVIEW

File: flutter/example/lib/main.dart [hunk 1]
```diff
@@ -98,6 +99,27 @@ class _MyAppState extends State<MyApp> {
     }
   }
 
+  Future<List<Wallet>> getWallets() async {
+    List<Wallet> wallets = await CoinbaseWalletSDK.shared.getWallets();
```
Comment (bangtoven) [inline flutter/example/lib/main.dart]: dart has static/class methods, right? can we refactor this to follow the design of ios and android?

**Review by hishboy — DISMISSED** [review]
The code looks great! Are you sure you've never written dart/flutter?

**Comment by cb-heimdall** [comment]
Review Error for hishboy @ 2022-11-30 18:53:44 UTC
User must have write permissions to review

**Comment by AMITGOELNY** [comment]
> The code looks great! Are you sure you've never written dart/flutter?

First timer, was pretty easy to pick up, and IDE support for dart is pretty good.

**Comment by bangtoven** [comment]
> The code looks great! Are you sure you've never written dart/flutter?

hey @hishboy thanks for chiming in!
we've recently refactored the SDK to support multi-wallet, which allows users to create an instance for individual wallet instead of using singleton. https://github.com/coinbase/wallet-mobile-sdk/pull/117/files
i see dart code had `CoinbaseWalletSDK.shared` seemingly a singleton object. 
how do you think about adopting the similar pattern to use a dedicated instance and deprecating the singleton? is this something we could do on flutter? cc @AMITGOELNY

File: flutter/android/src/main/kotlin/com/coinbase/flutter/wallet_sdk/CoinbaseWalletSdkFlutterPlugin.kt [hunk 2]
```diff
@@ -34,47 +35,40 @@ class CoinbaseWalletSdkFlutterPlugin : FlutterPlugin, MethodCallHandler,
...
-            if (call.method == "isAppInstalled") {
-                return isAppInstalled(result)
+            when (call.method) {
+                "configure" -> return configure(call, result)
+                "initiateHandshake" -> return initiateHandshake(call, result)
+                "makeRequest" -> return makeRequest(call, result)
+                "resetSession" -> return resetSession(result)
+                "isAppInstalled" -> return isAppInstalled(result)
+                "getWallets" -> return getWallets(result)
+                "connectWallet" -> return connectWallet(call, result)
```
Comment (bangtoven) [inline flutter/android/src/main/kotlin/com/coinbase/flutter/wallet_sdk/CoinbaseWalletSdkFlutterPlugin.kt]: please apply the changes to kotlin. https://github.com/coinbase/wallet-mobile-sdk/pull/154/files

File: flutter/android/src/main/kotlin/com/coinbase/flutter/wallet_sdk/CoinbaseWalletSdkFlutterPlugin.kt [hunk 3]
```diff
@@ -34,104 +35,115 @@ class CoinbaseWalletSdkFlutterPlugin : FlutterPlugin, MethodCallHandler,
...
-    private fun resetSession(@NonNull result: Result) {
-        coinbase.resetSession()
+    private fun resetSession(call: MethodCall, result: Result) {
+        val jsonString = call.arguments
+        if (jsonString !is String) {
+            return result.error("resetSession", "Missing args", null)
+        }
+
+        val decoded = Json.decodeFromString<InstanceMethodArgument<NoArgument>>(jsonString)
+        val mwpClient = CoinbaseWalletSDK.getClient(decoded.wallet)
```
Comment (bangtoven) [inline flutter/android/src/main/kotlin/com/coinbase/flutter/wallet_sdk/CoinbaseWalletSdkFlutterPlugin.kt:129]: would be nice to dedup this

File: flutter/android/src/main/kotlin/com/coinbase/flutter/wallet_sdk/InstanceMethodArgument.kt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class InstanceMethodArgument<T> {
+    val wallet: Wallet
+    val argument: T?
+}
+
+@Serializable
+data class NoArgument {}
```
Comment (bangtoven) [inline flutter/android/src/main/kotlin/com/coinbase/flutter/wallet_sdk/InstanceMethodArgument.kt:10]: not sure this is correct. couldn't run it locally

File: flutter/android/src/main/kotlin/com/coinbase/flutter/wallet_sdk/CoinbaseWalletSdkFlutterPlugin.kt (hunk 3 above)
Comment (bangtoven) [reply flutter/android/src/main/kotlin/com/coinbase/flutter/wallet_sdk/CoinbaseWalletSdkFlutterPlugin.kt:129]: e.g. 
image

**Review by CharliesAngel1 — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for CharliesAngel1 @ 2023-01-22 02:44:17 UTC
User must have write permissions to review

**Pull request closed without merging**